src/main/webapp/app/exercises/programming/manage/instructions-editor/analysis/programming-exercise-instruction-analysis.service.ts

```diff
@@ -113,19 +107,19 @@ export class ProgrammingExerciseInstructionAnalysisService {
      * @param analysis arbitrary number of analysis objects to be merged into one.
      */
     private mergeAnalysis = (...analysis: Array<AnalysisItem[]>) => {
-        return compose(
-            reduce((acc, [lineNumber, values, issueType]) => {
-                const lineNumberValues = acc[lineNumber];
-                const issueValues = lineNumberValues ? lineNumberValues[issueType] || [] : [];
-                return { ...acc, [lineNumber]: { ...lineNumberValues, [issueType]: [...issueValues, ...values] } };
-            }, {}),
-            map(([lineNumber, values, issueType]: AnalysisItem) => [
+        const reducer = (acc: ProblemStatementAnalysis, [lineNumber, values, issueType]: AnalysisItem) => {
+            const lineNumberValues = acc[lineNumber];
+            const issueValues = lineNumberValues ? lineNumberValues[issueType] || [] : [];
+            return { ...acc, [lineNumber]: { ...lineNumberValues, [issueType]: [...issueValues, ...values] } };
+        };
+        return analysis
+            .flat()
+            .map(([lineNumber, values, issueType]: AnalysisItem) => [
                 lineNumber,
                 values.map((id) => this.translateService.instant(this.getTranslationByIssueType(issueType), { id })),
                 issueType,
-            ]),
-            flatten,
-        )(analysis);
+            ])
+            .reduce(reducer, []);
```

Comment:
Your default here is an array but was an object

Reply:
`ProblemStatementAnalysis` is defined as `Array<{…}>`. Therefore, accumulating in an array is the correct approach.

Reply:
kk, so it was wrong before, correct?

Reply:
I think so, yes.